addons/binding/org.openhab.binding.mihome/ESH-INF/thing/sensor_switch.xml

```diff
@@ -0,0 +1,42 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<thing:thing-descriptions bindingId="mihome" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	xmlns:thing="http://eclipse.org/smarthome/schemas/thing-description/v1.0.0"
+	xsi:schemaLocation="http://eclipse.org/smarthome/schemas/thing-description/v1.0.0 http://eclipse.org/smarthome/schemas/thing-description-1.0.0.xsd">
+
+	<thing-type id="sensor_switch">
+		<supported-bridge-type-refs>
+			<bridge-type-ref id="bridge" />
+		</supported-bridge-type-refs>
+
+		<label>Xiaomi Mi Wireless Switch</label>
+		<description>Round-shaped mini wireless switch that allows to trigger four types of event: single click, double click, long click press, long click release.</description>
+
+		<channels>
+			<channel id="button" typeId="button" />
+			<channel id="batteryLevel" typeId="system.battery-level" />
+			<channel id="lowBattery" typeId="system.low-battery" />
+		</channels>
+
+		<config-description>
+			<parameter name="itemId" type="text" readOnly="true">
+				<label>Item ID</label>
+				<description>An identifier for the item connected.</description>
+				<required>true</required>
+			</parameter>
+		</config-description>
+	</thing-type>
+
+	<channel-type id="button">
```

Comment:
where is the point in providing the LONG_PRESSED event in SYSTEM_BUTTON then?
I looked it up, its two lines of code. And having that, there is still room for the raw button, which is not able to distinguish different clicks. for example to build own recognizers for three clicks or patterns etc.
would you like to have a PR on the ESH repo?
otherwise I would like to stay with the custom channels for devices which send LONG_RELEASED events.